src/components/webcams/WebrtcRTSPSimpleServer.vue

```diff
@@ -0,0 +1,251 @@
+<style scoped>
+.webcamImage {
+    width: 100%;
+}
+</style>
+
+<template>
+    <video
+        ref="video"
+        v-observe-visibility="visibilityChanged"
+        :style="webcamStyle"
+        class="webcamImage"
+        autoplay
+        playsinline
+        muted
+        controls />
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Ref } from 'vue-property-decorator'
+import BaseMixin from '@/components/mixins/base'
+
+@Component
+export default class WebrtcRTSPSimpleServer extends Mixins(BaseMixin) {
+    private isVisible = true
+
+    @Prop({ required: true })
+    camSettings: any
+
+    @Prop()
+    printerUrl: string | undefined
+
+    @Ref()
+    declare video: HTMLVideoElement
+
+    // webrtc player vars
+    private webRtcTerminated: boolean = false
+    private webRtcWs: any
+    private webRtcPc: any
+    private webRtcRestartTimeout: any
+
+    mounted() {
+        this.video.onload = () => {
+            if (this.webRtcTerminated) {
+                this.terminateWebRtcVideo() // catch a potential race condition
+            } else {
+                this.video.play() // trigger autoplay for some browsers
+            }
+        }
+    }
+
+    get webcamStyle() {
+        let transforms = ''
+        if ('flipX' in this.camSettings && this.camSettings.flipX) transforms += ' scaleX(-1)'
+        if ('flipX' in this.camSettings && this.camSettings.flipY) transforms += ' scaleY(-1)'
+        if (transforms.trimLeft().length) return { transform: transforms.trimLeft() }
+
+        return ''
+    }
+
+    isWebRtcUrl(): boolean {
+        return this.camSettings.urlStream.startsWith('ws://') || this.camSettings.urlStream.startsWith('wss://')
+    }
+
+    terminateWebRtcVideo() {
+        try {
+            this.video.pause()
+        } catch (err) {
+            // ignore - more important to shut down the sockets.
+        }
+
+        this.webRtcTerminate()
+    }
+
+    visibilityChanged(isVisible: boolean) {
```

Comment:
i would not use the visibilty check for webRTC, because webRTC needs too long to reconnect and it also need only a small bandwidth to work.